docs/5.0/preview/database-access.md

```diff
@@ -44,23 +44,82 @@ The following diagram shows an example Database Access setup:
+### On-Prem PostgreSQL
 
-We will create a Teleport cluster and connect a PostgreSQL-flavored AWS Aurora
-database to it via Teleport database proxy.
+!!! note
+    This section explains how to configure an on-premise instance of PostgreSQL
+    to work with Teleport Database Access. For information about configuring
+    AWS RDS/Aurora see the [section below](#aws-rdsaurora-postgresql).
 
-### Setup Aurora
+#### Create Certificate/Key Pair
 
-Teleport Database Access uses IAM authentication with AWS RDS and Aurora
-databases which can be enabled with the following steps.
+Teleport uses mutual TLS authentication so PostgreSQL server must be configured
+with Teleport's certificate authority and certificate/key pair that Teleport
+can validate.
+
+To create these secrets, use `tctl auth sign` command. Note that it requires a
+running Teleport cluster and [should be run](https://goteleport.com/teleport/docs/architecture/overview/#tctl)
+on the auth server.
+
+```sh
+$ tctl auth sign --format=db --host=db.example.com --out=server --ttl=8760h
+```
+
+Flag descriptions:
+
+* `--format=db`: instructs the command to produce secrets in the format suitable
+  for configuring a database server.
+* `--host=db.example.com`: server name to encode in the certificate, should
+  match the hostname Teleport will be connecting to the database at.
+* `--out=server`: name prefix for output files.
+* `--ttl=8760h`: certificate validity period.
+
+The command will create 3 files: `server.cas` with Teleport's certificate
+authority and `server.crt`/`server.key` with generated certificate/key pair.
+
+#### Configure PostgreSQL Server
+
+To configure PostgreSQL server to accept TLS connections, add the following
+to PostgreSQL configuration file `postgresql.conf`:
+
+```conf
+ssl = on
+ssl_cert_file = '/path/to/server.crt'
+ssl_key_file = '/path/to/server.key'
+ssl_ca_file = '/path/toa/server.cas'
+```
+
+See [Secure TCP/IP Connections with SSL](https://www.postgresql.org/docs/current/ssl-tcp.html)
+in PostgreSQL documentation for more details.
+
+Additionally, PostgreSQL should be configured to require client certificate
+authentication from clients connecting over TLS. This can be done by adding
+the following entries to PostgreSQL host-based authentication file `pg_hba.conf`:
+
+```conf
+hostssl all             all             ::/0                    cert
+hostssl all             all             0.0.0.0/0               cert
+```
+
+See [The pg_hba.conf File](https://www.postgresql.org/docs/current/auth-pg-hba-conf.html)
+in PostgreSQL documentation for more details.
+
+### AWS RDS/Aurora PostgreSQL
+
+!!! note
+    This section explains how to configure a PostgreSQL-flavored instance of
+    AWS RDS or Aurora database to work with Teleport Database Access. For
+    information about configuring an on-prem PostgreSQL see the [section above](#on-prem-postgresql).
+
+Teleport Database Access for AWS RDS and Aurora uses IAM authentication which
```

Comment:
We should add a note below about how to obtain the ARN for the IAM policy.

Reply:
Done.